pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -223,14 +223,21 @@ type cpuAccumulator struct {
 	result cpuset.CPUSet
 
 	numaOrSocketsFirst numaOrSocketsFirstFuncs
+
+	// 'options' is the StaticPolicyOptions. This is used to control the cpu sorting result.
+	// The sequence of returned CPU IDs depends on the policy.
+	// By default, cpus is sorted by sortAvailableCPUs()
+	// If staticPolicy is `SpreadPhysicalCPUsPreferredOption`, cpu is sorted by sortAvailableNonConnectedCPUs()
+	options StaticPolicyOptions
 }
 
-func newCPUAccumulator(topo *topology.CPUTopology, availableCPUs cpuset.CPUSet, numCPUs int) *cpuAccumulator {
+func newCPUAccumulator(topo *topology.CPUTopology, availableCPUs cpuset.CPUSet, numCPUs int, opts StaticPolicyOptions) *cpuAccumulator {
```

Comment:
can we infer the packed vs spread allocation request or do we need to pass a new flag? truth to be told I'd just pass a boolean (or perhaps even the function pointer) down the chain vs the full options struct. Thing is, the options struct is a mixed bag and passing around data which doesn't belong here doesn't feel fret, even though is going to be harmless. But I'd minimize the data nevertheless (or repack it in a better way into a helper struct)

Reply:
Currently, I feel we have to pass a flag there. I agree the options looks a little bit weird. I will change to boolean or function pointer. Let me polish the code

Reply:
`numaOrSocketsFirst` is inferred from the topology but packed vs spread allocation is from user configuration, so infer can not work here as I know.